File: ProjetoModulo01/Elfo.java
public class Elfo {
    private String nome;
    private Integer flechas;
    private int experiencia = 0;
    
    public Elfo(String n, Integer flech) {
        nome = n;
        flechas = flech;
        if(flech == null){
            flechas = 42;
        }
    }
    
    public int getFlechas(){
        return flechas;
    }
    
    public void setFlechas(int flechas){
        this.flechas = flechas;
    }
    
    public void atirarFlecha(){
        experiencia++;
    }
}
    
    
    
    
   

File: src/modulo-03-web/dia-04/Elfo.js
function Elfo(options) {
  options = options || {};
  this.nome = options.nome;
  this.flechas = options.flechas || 42;

  Object.defineProperty(this, 'flechas', {
    writable: false
  });
}

Elfo.prototype.atirarFlecha = function(dwarf) {
  console.log('atirando flecha...');
};